### PR TITLE
Merge Toggle Map with Main 

### PR DESCRIPTION
**Pull request by kdumais111**
Heat map changes ready to merge

### CODE REVIEW

File: app/route_rangers_api/views.py [hunk 1]
```diff
@@ -236,3 +256,48 @@ def get_survey_context(city, form):
         "form": form,
     }
     return context
+
+# from django.db.models import Sum
```
Comment (lisette-solis) [inline app/route_rangers_api/views.py:280]: Is this needed?

**Review by lisette-solis — CHANGES_REQUESTED** [review]

File: app/route_rangers_api/static/heatmap.js [hunk 2]
```diff
@@ -2,79 +2,198 @@ function heatmaps(
...
-
+  // Initialize the map
   var heatmap = L.map("heatmap").setView(coordinates, 9);
-  var geojson;
 
+  // Add base layer
   var tiles = L.tileLayer(
     "https://{s}.basemaps.cartocdn.com/light_all/{z}/{x}/{y}{r}.png",
     {
       maxZoom: 19,
```
[the remark this answers is not in the document]
Comment (kdumais111) [reply app/route_rangers_api/static/heatmap.js]: I think larger issue later

File: app/route_rangers_api/static/heatmap.js [hunk 3]
```diff
@@ -2,79 +2,198 @@ function heatmaps(
...
+
+    info.addTo(heatmap);
+    infoControls[variable] = info;
+  });
+
+  // Hide all info boxes initially
+  Object.values(infoControls).forEach(info => info._div.style.display = 'none');
+  infoControls[currentVariable]._div.style.display = 'block';
+
+  // Add legend control
```
[the remark this answers is not in the document]
Comment (kdumais111) [reply app/route_rangers_api/static/heatmap.js:108]: done

File: app/route_rangers_api/views.py (hunk 1 above)
Comment (kdumais111) [reply app/route_rangers_api/views.py]: deleted

**Comment by kdumais111** [comment]
@meganhmoore once more with feeling?

**Review by meganhmoore — APPROVED** [review]
looks great, thanks for making all the edits

**Pull request merged**